src/ring_theory/ring_hom_properties.lean

```diff
@@ -119,6 +120,33 @@ begin
   exact H,
 end
 
+variables {R S R' S' : Type u} [comm_ring R] [comm_ring S] [comm_ring R'] [comm_ring S']
+variables [algebra R R'] [algebra R S'] [algebra S S'] [algebra R S] [is_scalar_tower R S S']
+
+lemma stable_under_base_change.base_change (hP : ring_hom.stable_under_base_change @P)
+    (hP' : ring_hom.respects_iso @P) (f : R' →ₐ[R] S') (hf : is_base_change S f.to_linear_map)
```

Comment:
I think that `respects_iso` ought not be present here. The fact that you need it, suggests to me that `stable_under_base_change` has a definition that is too restrictive.

Reply:
I don't think it matters both practically and morally since "properties" should always be invariant under isomorphisms (of arrows), and we should always have a `respects_iso` proof at hand.
Besides, this pattern is also followed in `ring_hom.of_localization_span` and `morphism_prolerty.stable_under_base_change` where the property is only asserted on "the" localization or "the" pullback, and the general cases follow from it and `respects_iso`.

I could still make this the definition (which would imply `respects_iso` on its own), and make this an auxiliary constructor instead if you think that is better.

Reply:
> I could still make this the definition (which would imply `respects_iso` on its own), and make this an auxiliary constructor instead if you think that is better.

I think that would be a good idea. Because I think it will be easier to use that way.